game/src/sandbox/speed.rs

```diff
@@ -156,34 +159,132 @@ impl TimePanel {
         self.panel = panel.build(ctx);
     }
 
-    fn create_time_panel(&mut self, ctx: &EventCtx, app: &App) -> Widget {
+    fn trips_completion_bar(&mut self, ctx: &EventCtx, app: &App) -> Widget {
+        let text_color = Color::WHITE;
+        let bar_fg = ctx.style().primary_fg;
+        let bar_bg = bar_fg.tint(0.6).shade(0.2);
+        let cursor_fg = Color::hex("#939393");
+
+        // This is manually tuned
+        let bar_width = 400.0;
+        let bar_height = 27.0;
+
         let (finished, unfinished) = app.primary.sim.num_trips();
-        let trip_results = Widget::row(vec![
-            {
-                let mut txt = Text::new();
-                let pct = if unfinished == 0 {
-                    100.0
+        let total = finished + unfinished;
+        let ratio = if total > 0 {
+            finished as f64 / total as f64
+        } else {
+            0.0
+        };
+        let finished_width = ratio * bar_width;
+
+        if app.has_prebaked().is_some() {
+            let now = self.time;
+            let mut baseline_finished = self.baseline_finished_trips.unwrap_or(0);
+            for (t, _, _, _) in &app.prebaked().finished_trips[baseline_finished..] {
```

Comment:
Ahhh, clever trick. Most of the things that read `Analytics` recalculate everything from scratch constantly. To one day speed this up, I was imagining making some kind of stateful "cursor" objects to remember where a computation left off. But that's total overkill for something like this -- just remembering the count is much simpler.